neurons/miner.py

```diff
@@ -100,6 +100,17 @@ async def blacklist(
 
         Otherwise, allow the request to be processed further.
         """
+        if not synapse.dendrite.hotkey:
+            return True, "Hotkey not provided"
+        registered = synapse.dendrite.hotkey in self.metagraph.hotkeys
+        if self.config.blacklist.allow_non_registered and not registered:
+            return False, "Allowing un-registered hotkey"
+        elif not registered:
+            bt.logging.trace(
+                f"Blacklisting un-registered hotkey {synapse.dendrite.hotkey}"
+            )
+            return True, f"Unrecognized hotkey {synapse.dendrite.hotkey}"
+
         uid = self.metagraph.hotkeys.index(synapse.dendrite.hotkey)
         if (
```

Comment:
@jondurbin - don't current lines 104-112 already handle what your proposed lines of 105-113 do?

Maybe we just need 103-104?

Reply:
Oops I forgot to remove the old block - it's in there now.  The issue is this line:
```
uid = self.metagraph.hotkeys.index(synapse.dendrite.hotkey)
```
And fetching items based on that uid.  Basically any time the hotkey is invalid or missing, the case must be handled before the `.index(hotkey)` call.

Reply:
Awesome, thank you.